project_euler/problem_079/sol1.py

```diff
@@ -0,0 +1,53 @@
+"""
+Project Euler Problem 79: https://projecteuler.net/problem=79
+
+Passcode derivation
+
+A common security method used for online banking is to ask the user for three
+random characters from a passcode. For example, if the passcode was 531278,
+they may ask for the 2nd, 3rd, and 5th characters; the expected reply would
+be: 317.
+
+The text file, keylog.txt, contains fifty successful login attempts.
+
+Given that the three characters are always asked for in order, analyse the file
+so as to determine the shortest possible secret passcode of unknown length.
+...
+
+"""
+import itertools
+import os
+
+
+def solution() -> int:
+    """
+    Returns the shortest possible secret passcode of unknown length.
+
+    >>> solution()
+    73162890
```

Comment:
As mentioned in the [README](https://github.com/TheAlgorithms/Python/tree/master/project_euler#coding-style):

> There should not be a doctest for testing the answer as that is done by our GitHub Actions build ...

```suggestion
```

Instead, have some test cases. Now as this requires reading a file to get the input, you could separate out the core part of the solution (which is everything after the file is read) and take in a parameter `logins`:

```python
# This function could be tested with variety of inputs.
def find_secret_passcode(logins):
	...


def solution():
	# Read the keylog.txt file and fill the `logins` array
    logins = [...]

	return find_secret_passcode(logins)
```

Reply:
Good idea. I will implement this.